pkg/util/annotation.go

```diff
@@ -17,8 +25,16 @@ func MergeAnnotation(obj *unstructured.Unstructured, annotationKey string, annot
 
 // MergeAnnotations merges the annotations from 'src' to 'dst', identical keys will not be merged.
 func MergeAnnotations(dst *unstructured.Unstructured, src *unstructured.Unstructured) {
+	objectAnnotation := dst.GetAnnotations()
+	if objectAnnotation == nil {
+		objectAnnotation = make(map[string]string, 0)
+	}
+	deletedAnnotationKeys := getDeletedAnnotationKeys(dst, src)
 	for key, value := range src.GetAnnotations() {
-		MergeAnnotation(dst, key, value)
+		if _, exist := objectAnnotation[key]; !exist && !deletedAnnotationKeys.Has(key) {
+			objectAnnotation[key] = value
+			dst.SetAnnotations(objectAnnotation)
```

Comment:
We can move this outside the `for` loop.

Reply:
No we can't, if `src.GetAnnotations()` is `nil` , `dst.SetAnnotations` will set an empty map .